### PR TITLE
create notebook examples

### PR DESCRIPTION
**Pull request by colombod**
This PR contains notebooks to show functionalities of the Kernel and interesting applications. It relates to #408

### CODE REVIEW

**Comment by colombod** [comment]
> I followed the samples on my pc and get these convert type errors from 'XPlot.Plotly.Graph.Scatter[]' to 'XPlot.Plotly.Graph.Trace'
> image
> Wasn't the result meant to be convert from plot to histogram?

The problem here is related to type resolution bug that is reported in #345 and dotnet/roslyn#23123

**Comment by LadyNaggaga** [comment]
> > I followed the samples on my pc and get these convert type errors from 'XPlot.Plotly.Graph.Scatter[]' to 'XPlot.Plotly.Graph.Trace'
> > image
> > Wasn't the result meant to be convert from plot to histogram?
> 
> The problem here is related to type resolution bug that is reported in #345 and [dotnet/roslyn#23123](https://github.com/dotnet/roslyn/issues/23123)

Oh okay. Thoughts on removing this section before or linking to the issue in the sample ?

File: NotebookExamples/csharp/Kernel features.ipynb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
```
Comment (LadyNaggaga) [inline NotebookExamples/csharp/Kernel features.ipynb]: This samples looks good to share.  Just need to make a tiny typo colelctions to collections.
image

File: .gitignore [hunk 2]
```diff
@@ -373,3 +373,5 @@ docs/\.ionide/
 \.ionide/
 
 
+
+NotebookExamples/csharp/.ipynb_checkpoints/
```
Comment (eerhardt) [inline .gitignore]: Would it make more sense to ignore any `.ipynb_checkpoints` directory?

See
[https://github.com/github/gitignore/blob/eb14c04c3596082f0d16a543d54b1fe1608c6498/community/Python/JupyterNotebooks.gitignore#L5](https://github.com/github/gitignore/blob/eb14c04c3596082f0d16a543d54b1fe1608c6498/community/Python/JupyterNotebooks.gitignore#L5)
[https://github.com/github/gitignore/blob/c1faae44aebe587bbb90d60e21ca928e0116e8d4/Python.gitignore#L77](https://github.com/github/gitignore/blob/c1faae44aebe587bbb90d60e21ca928e0116e8d4/Python.gitignore#L77)

File: FSharpWorkspaceShim/FSharpWorkspaceShim.fsproj [hunk 3]
```diff
@@ -18,7 +18,7 @@
 
   <ItemGroup>
     <PackageReference Include="FSharp.Compiler.Service" />
-    <PackageReference Include="Microsoft.CodeAnalysis" Version="3.2.1" />
+    <PackageReference Include="Microsoft.CodeAnalysis" Version="3.4.0-beta2-19467-02" />
```
Comment (eerhardt) [inline FSharpWorkspaceShim/FSharpWorkspaceShim.fsproj]: Is there a central place you can put these version numbers, so they don't have to be updated in so many places?

File: NotebookExamples/csharp/Repo Statistics.ipynb [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+       "}\r\n",
+       "</script>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "//display(createdThisMonth.Select(i => new {i.CreatedAt, i.Title, State = i.State.StringValue,  i.Number}));\n",
```
Comment (eerhardt) [inline NotebookExamples/csharp/Repo Statistics.ipynb]: Is this comment needed?

File: NotebookExamples/csharp/Repo Statistics.ipynb [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+       "}\r\n",
+       "</script>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "//display(openByMonth);\n",
```
Comment (eerhardt) [inline NotebookExamples/csharp/Repo Statistics.ipynb]: Can this comment be removed?

**Pull request merged**